src/plugins/kibana_react/public/util/to_mount_point.tsx

```diff
@@ -8,17 +8,27 @@
 
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { Observable } from 'rxjs';
 import { I18nProvider } from '@kbn/i18n/react';
-import { MountPoint } from 'kibana/public';
+import type { MountPoint, CoreTheme } from 'kibana/public';
+import { KibanaThemeProvider } from '../theme';
+
+export interface ToMountPointOptions {
+  theme$?: Observable<CoreTheme>;
+}
 
 /**
  * MountPoint converter for react nodes.
  *
  * @param node to get a mount point for
  */
-export const toMountPoint = (node: React.ReactNode): MountPoint => {
+export const toMountPoint = (
+  node: React.ReactNode,
+  { theme$ }: ToMountPointOptions = {}
+): MountPoint => {
+  const content = theme$ ? <KibanaThemeProvider theme$={theme$}>{node}</KibanaThemeProvider> : node;
   const mount = (element: HTMLElement) => {
-    ReactDOM.render(<I18nProvider>{node}</I18nProvider>, element);
+    ReactDOM.render(<I18nProvider>{content}</I18nProvider>, element);
```

Comment:
I only added the `theme$: Observable<CoreTheme>` option for now. The reasoning is that `toMountPoint` is usually used outside of a react tree, like directly from an application or a service, so I think consumers will more easily have access to core's contract to retrieve the theme, than passing the eui theme accessed from a component via `useEuiTheme`.

Similar to previous comment, this can be added easily after, so we may want to wait for our consumers usages before going further.